executor/runtime/docker/docker.go

```diff
@@ -1887,58 +1891,115 @@ func handleDockerEvent(c runtimeTypes.Container, message events.Message, statusM
 		}
-		return false
+		return nonTerminalDockerEvent
 	case "kill":
-		statusMessageChan <- runtimeTypes.StatusMessage{
-			Status: runtimeTypes.StatusFailed,
-			Msg:    fmt.Sprintf("main container killed with signal %s", message.Actor.Attributes["signal"]),
+		// TODO: Handle the difference between platform/user sidecar, not all
+		// "kills"s should result in a Failed
+		if cName == mainContainerName {
+			statusMessageChan <- runtimeTypes.StatusMessage{
+				Status: runtimeTypes.StatusFailed,
+				Msg:    fmt.Sprintf("%s container killed with signal %s", cName, message.Actor.Attributes["signal"]),
+			}
+		} else {
+			l.Debugf("Skipping docker kill event for %s, no need to update the pod", cName)
+			return nonTerminalDockerEvent
 		}
 	case "oom":
+		// TODO: Handle the difference between platform/user sidecar, not all
+		// "oom"s should result in a Failed
 		statusMessageChan <- runtimeTypes.StatusMessage{
 			Status: runtimeTypes.StatusFailed,
-			Msg:    fmt.Sprintf("main container %s exited due to OOMKilled", c.TaskID()),
+			Msg:    fmt.Sprintf("%s container %s exited due to OOMKilled", cName, r.c.TaskID()),
 		}
+		return isTerminalDockerEvent
 	// Ignore exec events entirely
 	case "exec_create", "exec_start", "exec_die":
-		return false
+		return nonTerminalDockerEvent
 	default:
-		log.WithField("taskID", c.ID()).Info("Received unexpected docker event: ", message)
-		return false
+		log.WithField("taskID", r.c.ID()).Info("Received unexpected docker event: ", message)
+		return nonTerminalDockerEvent
+	}
+
+	return nonTerminalDockerEvent
+}
+
+// getContainerNameFromDockerEvent pulls out the titus-friendly container name (whatever the user specified)
+// out of a docker event. Docker events look like this:
+//
+// {"status":"exec_die","id":"408563a253eb82bf0e76e3cd32594dc55a968a16dc494c45d8d46c7da465ce82",
+//  "from":"busybox","Type":"container","Action":"exec_die",
+//  "Actor":{
+//    "ID":"408563a253eb82bf0e76e3cd32594dc55a968a16dc494c45d8d46c7da465ce82",
+//    "Attributes":{
+//      "execID":"3514a0192401ad20c1bbf693d38d64e036b8ca2b700894cb1cea9df43dcfa34f","exitCode":"1","image":"busybox","name":"123-foobar"}
+//     },
+//   "scope":"local","time":1626119653,"timeNano":1626119653617364285
+// }
+//
+// For Titus, the id (container id) is the best way to correlate where this
+// event is coming from.
+func (r *DockerRuntime) getContainerNameFromDockerEvent(m events.Message) string {
+	return r.getContainerNameFromID(m.ID)
+}
+
+func (r *DockerRuntime) getContainerNameFromID(id string) string {
+	// Most common case, just the main container, we return the string "main"
+	if id == r.c.ID() {
+		return mainContainerName
+	}
+	r.c.ExtraPlatformContainers()
+	for _, c := range r.c.ExtraPlatformContainers() {
+		if id == c.Status.ContainerID {
+			return c.Name
+		}
 	}
-
-	return true
+	for _, c := range r.c.ExtraUserContainers() {
+		if id == c.Status.ContainerID {
+			return c.Name
+		}
+	}
+	return "Unknown container"
```

Comment:
This seems to be an error condition, perhaps return error here and log the entire message before marking it "unknown container"?